src/PhpSpec/Matcher/ScalarMatcher.php

```diff
@@ -127,6 +127,10 @@ private function getCheckerName(string $name)
             return false;
         }
 
+        if ($name == 'beReal') {
+            trigger_error('`beReal` matcher is deprecated, please use `beFloat`', E_USER_DEPRECATED);
```

Comment:
use `@trigger_error('...', E_USER_DEPRECATED);` to follow the opt-in deprecation reporting that the Symfony community defined (after breaking the whole ecosystem in 2.7.0 when trying to use it without silencing, for similar reasons than what you have here)

Reply:
Sorry, but how this is supposed to work? I trigger notice and immediately silence it. How the users will know that this matcher is deprecated?

Reply:
@stof OK I found something about it

Reply:
an error handler is still able to handle silenced errors, but they can know whether what they handle is silenced or no.

And well-behaving error handlers which are converting errors to exceptions would be doing it only for non-silenced errors, which is what makes the deprecation warning opt-in

Reply:
Thanks. I've fixed that. But I have no idea how to test this using PhpSpec. What about integration test?

Reply:
@stof can you review my changes again please? :)